pkg/kv/dynamodb/store.go

```diff
@@ -453,6 +421,59 @@ func (e *EntriesIterator) Close() {
 	e.err = kv.ErrClosedEntries
 }
 
+func (e *EntriesIterator) runQuery(exclusiveStartKey map[string]*dynamodb.AttributeValue) {
```

Comment:
It's kind of weird to have `exclusiveStartKey` and not `skipStartKey` which you have in a different field.